dsl/src/main/scala/com.crobox.clickhouse/dsl/JoinQuery.scala

```diff
@@ -57,7 +57,19 @@ object JoinQuery {
  */
 case class JoinQuery(joinType: JoinType,
                      other: FromQuery,
-                     //on: Seq[ExpressionColumn[_]] = Seq.empty,
-                     on: Seq[(Column, String, Column)] = Seq.empty,
+                     on: Seq[JoinCondition] = Seq.empty,
                      using: Seq[Column] = Seq.empty,
                      global: Boolean = false)
+
+case class JoinCondition(left: Column, operator: String, right: Column) {
+
+  assert(JoinCondition.SupportedOperators.contains(operator),
```

Comment:
`assert` or `require`?

Reply:
From StackOverflow:  https://stackoverflow.com/questions/24863185/what-is-an-assertionerror-in-which-case-should-i-throw-it-from-my-own-code
 
The meaning of an AssertionError is that something happened that the developer thought was impossible to happen.